src/gesture-handler/GestureDetector.tsx

```diff
@@ -0,0 +1,302 @@
+  swipeRight: ScreenTransition.SwipeRight,
+  swipeLeft: ScreenTransition.SwipeLeft,
+  swipeDown: ScreenTransition.SwipeDown,
+  swipeUp: ScreenTransition.SwipeUp,
+  horizontalSwipe: ScreenTransition.Horizontal,
+  verticalSwipe: ScreenTransition.Vertical,
+  twoDimensionalSwipe: ScreenTransition.TwoDimensional,
+};
+
+const EmptyGestureHandler = Gesture.Tap();
+
+const SupportedGestures = [
+  'swipeRight',
+  'swipeLeft',
+  'swipeDown',
+  'swipeUp',
+  'horizontalSwipe',
+  'verticalSwipe',
+  'twoDimensionalSwipe',
+];
+
+const DefaultEvent: GestureUpdateEvent<PanGestureHandlerEventPayload> = {
+  absoluteX: 0,
+  absoluteY: 0,
+  handlerTag: 0,
+  numberOfPointers: 0,
+  state: 0,
+  translationX: 0,
+  translationY: 0,
+  velocityX: 0,
+  velocityY: 0,
+  x: 0,
+  y: 0,
+};
+
+const DefaultScreenDimensions = {
+  width: 0,
+  height: 0,
+  x: 0,
+  y: 0,
+  pageX: 0,
+  pageY: 0,
+};
+
+const TransitionHandler = ({
+  children,
+  stackRef,
+  goBackGesture,
+  screenEdgeGesture,
+  transitionAnimation: userTransitionAnimation,
+}: GestureProviderProps) => {
+  if (stackRef === undefined) {
+    throw new Error('[RNScreens] You have to specify `stackRef`');
+  }
+  stackRef.current = useAnimatedRef();
+  const sharedEvent = useSharedValue(DefaultEvent);
+  const startingGesturePosition = useSharedValue(DefaultEvent);
+  const canPerformUpdates = useSharedValue(false);
+  let transitionAnimation = ScreenTransition.SwipeRight;
+  if (userTransitionAnimation) {
+    transitionAnimation = userTransitionAnimation;
+    if (!goBackGesture) {
+      throw new Error(
+        '[RNScreens] You have to specify `goBackGesture` when using `transitionAnimation`'
+      );
+    }
+  } else {
+    if (!!goBackGesture && SupportedGestures.includes(goBackGesture)) {
+      transitionAnimation = AnimationForGesture[goBackGesture];
+    } else if (goBackGesture !== undefined) {
+      throw new Error(`[RNScreens] Unknown goBackGesture: ${goBackGesture}`);
+    }
+  }
+  const screenTransitionConfig = makeMutable(
+    {
+      stackTag: -1,
+      belowTopScreenTag: -1,
+      topScreenTag: -1,
+      sharedEvent,
+      startingGesturePosition,
+      screenTransition: transitionAnimation,
+      isTransitionCanceled: false,
+      goBackGesture: goBackGesture ?? 'swipeRight',
+      screenDimensions: DefaultScreenDimensions,
+      onFinishAnimation: () => {
+        'worklet';
+      },
+    },
+    true
+  );
+  const refCurrent = stackRef.current;
+  if (refCurrent === null) {
+    throw new Error('[RNScreens] You have to specify `stackRefWrapper`');
+  }
+
+  function onStart(event: GestureUpdateEvent<PanGestureHandlerEventPayload>) {
+    'worklet';
+    sharedEvent.value = event;
+    const transitionConfig = screenTransitionConfig.value;
+    const animatedRef = refCurrent;
+    if (!animatedRef) {
+      throw new Error('[Reanimated] Unable to recognize stack ref.');
+    }
+    const stackTag = (animatedRef as () => number)();
```

Comment:
We need many more comments on what is going on in this code, or it will be impossible to maintain it in the future xd

Reply:
I removed it 😅